components/ws-manager-api/core.proto

```diff
@@ -124,6 +127,29 @@ enum StopWorkspacePolicy {
     ABORT = 2;
 }
 
+enum StopWorkspaceReason {
+    // The user stopped the workspace
+    USER_STOPPED = 0;
```

Comment:
0 index enum should always be an UNKNOWN value. This is because when a field is not set in proto/grpc, it defaults to the 0 value. For enums, that's the value with the 0 index. In this case, if our system forgot to set the reason, it would default to USER_STOPPED which is not what we want - we want it to be UNKNOWN.

Reply:
Thanks! Good idea (I also felt there was a missing "unknown" state here, although there is no need for it yet in the current code)

Reply:
There is on the protocol level however. For example, if ws-manager implemented using the StopWorkspaceReason before server started setting it, it would always get the USER_STOPPED. There's a proto lint rule that should've caught this, I'll need to investigate why it didn't trigger here.